### PR TITLE
Slime Core Reaction Expansion

### PR DESCRIPTION
**Pull request by DenkouZ**
This is a slime feature request with the idea of trying to make xenobiology more useful for the station.
There is a mix of recipes in here. 

There is one small buff I have changed in all of this. That is changing Charged Slime Core Cells from 10000 charge to 30000 charge. This make the time spent gathering the core more useful as for before. The 30 or so minutes it might take to mutate a slime core that had the equivalent charge of a High Capacity Power Cell felt mostly useless as by that point the rest of science would be producing Super and Hyper cells. The rest of the changes are all expansions on the slime core reactions.

The intention behind these changes it to make the xenobiologist require more than just plasma to operate and provide for the station, also the ingredients chosen are to make the Xenobiologist leave the lab to acquire them or to collaborate with someone else in the science, chemistry or mining departments.
Although some slimes take longer than others to obtain, some of the earlier mutations have some more functionality. This is to make the xenobiologist make decisions about what slimes he is to keep and what ones to continue breeding.

_**There have been 3 new potions added to the slime reaction table.**_

**Slime Resurrection Serum**: This potion is to be used on unused slime cores to make them into child slimes. They are not loyal to anyone as they are new slimes but this offers a way of storing slimes in the smartfridge and also offers a way to bring back killed slimes if something does occur in the lab.
Also added while making this item was adding a primary type to all of the slime cores so that BYOND can read what kind of slimes they come from. Maybe someone with more imagination than I can come up with interesting ideas using this variable.

**Slime Duplicator**: This is a potion that can be used on a baby slime causing it to split in two giving you 2 child slimes. The intent behind this is to be able to make quick splits from your child mutations if you want an extract quickly but can also serve the purpose of giving you another slime to be able to try and mutate onto the next stage.

**Slime Nutrient**: This potion is a method of quickly growing slimes if you have the resources and the cores for it. It comes with two uses in the bottle. This is to increase the variety in what you can apply it in the lab. You can fully grow and split a baby slime for a chance at a quick mutation. You can grow two baby slimes into adults to save some time and extending from that you can use it to split off 2 adult slimes.

_**Next is all Slime core reactions and expanding on what they do**_. While reading over these I would like to keep in mind that slime cores are produced rather slowly and XenoBiology isn't turning out tens of cores in half an hour. Also that if some of these seem small or insignificant it is because the intent of some of these ideas it to be a crutch for science if other departments need the product or if science finds their stocks lacking.

**Green slime cores** 
Now have the ability to produce more chems.
+water = 5u bottle of Peridoxin
+oxygen = 5u bottle of Dexalin Plus
+sugar = 10u bottle of Doctor's Delight
+carbon = 10u bottle of Bicaridine
+phosphorus = 5u bottle of Dermaline 
These reactions have been set up to be able to drip feed some chems out of XenoBio if someone wants to produce them but the main feature of this is another way for people to acquire Peridoxin. It is a very round about method to get the chemical. But with this option on the table it means that it is available from more than just borers.

**Metal slime cores**
Can now make more materials
+Copper = 5 sheets of Gold
+Tungsten = 5 sheets of Silver
+Radium = 5 sheets of Uranium
+Carbon = 2 sheets of Diamond
The intention behind these reactions is to give RnD a slow feed of materials if they need it. These reactions are enough to get some items produced but not enough to be reliable on still keeping a strong cargo presence in Science. There are some shifts where things don't pan out for miners and they can't find diamond or uranium is scattered and hard to find. This is a way to help support both departments after a time of waiting on slimes to mutate and being processed. 

**Silver slime cores**
Have a new function.
+Carbon =  Creates 2 stacks of 5 sheets of a random material. (The pool for these is : Metal, Plasma, Gold, Uranium)
The idea behind this is just a small upgrade from metal slimes to be able to produce more materials from a single slime core while still keeping in theme of the silver slime's randomized reactions.

**Dark blue slime cores**
Have a reaction to create Slime Nutrient
+blood = One bottle with 2 uses of Slime Nutrient

 **Red slime cores**
Can create the Slime Resurrection Serum now
+blood = One bottle with 1 use of Slime Resurrection Serum

**Black slime cores**
Now hold a new reaction to be able to make NanoMachines from a recent addition.
+Gold = 1u Medical Nanomachines that can be drawn from the used extract
+Uranium = 1u Combat Nanomachines that can be drawn from the used extract.
The intent behind this is to give another method to acquire Nanomachines late into slime mutations. With this said it is a very small amount, that is so that people can not generate nanomachines quickly and suddenly start a cyber horror outbreak.

**Oil slime cores**
Can now use a new function
+blood = The slime genocide. Killing all slimes in view of the tile it is activated from. It will instantly kill regular slimes in their tracks and force adult slimes to split into 2 rabid slimes. 
This is made to have 2 functions in mind. Mass culling or starting a Rabid Slime outbreak. Or killing off an outbreak of slimes from a wild outbreak in the library or if wormholes suck all of the slimes out of containment

**Adamantine slime cores**
Have an advanced version of the metal slime reaction
+carbon = 5 sheets of diamond
+silver = 5 sheets of bananium
+gold = 5 sheets of phazon
These slimes are very late game and these reactions are to inspire more people to head towards these options. With telescience not being able to find the clown roid to get bananium or phazon and space travel becoming much more longer for miners because of the increase in zlevel size this reaction has been made to give an alternative to finding these materials so robotics can have some fun making other big stompy mechs.

**Cerulean slime cores**
Make Slime Duplicator potions now.
+blood = One bottle with 1 use of Slime Duplicator

**Pyrite slime cores**
Have one small added feature
+blood = 100 space cash chip
Seeing how silver slimes act as a service slime with food and drink and pyrite slimes go into more fun things like paint. I figured another way to get cash out of slimes would be fun.

### CODE REVIEW

**Comment by DenkouZ** [comment]
I would also like to give credit where credit is due. Kammerjunk helped a lot with this code. Making the code for the Slime Res, Slime Duplicator as well as the Metal slime reactions, Silver slime reaction, Slime Peridoxin and the Slime Genocide reaction. This PR would not exist without Kammerjunks efforts and helping learn BYOND code.

**Comment by PJB3005** [comment]
:+1: to more xeno biology toys, the job has had little use other than reviving ghosts. 

One issue might be balance, though we'll have to see how that turns out.

File: code/modules/mob/living/carbon/slime/slime.dm [hunk 1]
```diff
@@ -938,6 +972,33 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
 		M.cores = 3
 		qdel (src)
 
+
+/obj/item/weapon/slimenutrient
+	name = "Slime Nutrient"
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/slime/slime.dm]: Lowercase please.

File: code/modules/mob/living/carbon/slime/slime.dm [hunk 2]
```diff
@@ -938,6 +972,33 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
...
+	desc = "A potent chemical mix that is a great nutrient for slimes."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle12"
+	var/Uses = 2
+
+/obj/item/weapon/slimenutrient/attack(mob/living/carbon/slime/M as mob, mob/user as mob)
+	if(!istype(M, /mob/living/carbon/slime))//If target is not a slime.
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/slime/slime.dm]: You can leave out the second argument of the `istype` because BYOND knows what type of variable `M` is.

File: code/modules/mob/living/carbon/slime/slime.dm [hunk 3]
```diff
@@ -957,6 +1018,36 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
 			target.enahnced = 1
 			del (src)*/
 
+/obj/item/weapon/slimedupe
+    name = "slime duplicator"
+    desc = "A potent chemical mix that will force a child slime to split in two!"
+    icon = 'icons/obj/chemical.dmi'
+    icon_state = "bottle15"
+
+    attack(mob/living/carbon/slime/M as mob, mob/user as mob)
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/slime/slime.dm]: You should absolutely path this proc.

**Comment by icantthinkofanameritenow** [comment]
Hold onto your hat, this is going to be a long one. 

I don't think you understand exactly how powerful peridaxon is. With this PR, the chem that makes surgery completely unnecessary can now be made consistently instead of either jumping through Cargo's to find the right exotic seed or getting borers to show up. Contrary to what you claim, it's far from roundabout by comparison. (And ironically, given how quickly chem dispensers are made in RnD nobody would even touch the other chems it can make. Yes, even the nanomachines since those can also be mass produced with said chem.dispensers. Technically they don't even to make it themselves, they can just take the one in the research outpost.)

Additionally, the "extra options" for making materials fall flat on their faces when you realize that even a semi-competent miner cam have RnD completely set for materials less than 15 minutes into the round while you'd still be praying to the RNG  for the slimes to mutate into what you want. Even if there were no miners or mining borgs in a given round,  the material synthesizer that RnD can make (or just bug an engiborg into using for them) will make it so they need only one sheet of the material and then will have nothing to do with you. With all those superior options already present, the people who would still use the drip-feeding options are those who would be doing Xenobio no matter what you reward them with.

As for slime duplicator and slime nutrient, I would say that it would be easier to directly change the slime code so they need to be fed less to grow and split. Same overall result, without wasting time and slime cores. 

I said it once in the thread, and I shall say it again- you're looking at the problem all wrong. Instead of having xenobio be an unnecessarily complex method of generating laughably low amounts of resources, it should make things that are completely new and otherwise very difficult if not outright impossible to get.

Take black food coloring, for example- there is literally no other way to make.more after exhausting the supply the bartender starts with but isn't nearly valuable enough to justify that rarity. It's a logical choice for something Xenobio could produce. Same goes for things like universal enzyme (impossible to get more without Cargo) or minerals that have no normal way to acquire right now like mythril and adamantine. Alternatively, make the existing stuff that gets made in the reactions be more useful. For example, slime jelly actually has a small chance to heal organ damage, and all it would take is a little tweaking of the numbers to make that chance high enough to be worth gambling over-a 10-20% probability sounds acceptable to me. Surely you can think of other things too, right?

TL;DR: instead of making things that can be acquired through better methods, have slime cores produce entirely new things people will want.

**Comment by icantthinkofanameritenow** [comment]
@PJB3005

And when did every job need a specific use? Look at Xenoarchaeology, which consistently produced trash almost all the time with the few exceptions being lethal more often than not. Or the IAA, whose main role is to send faxes to CentComm that never get answered. Or the Mechanic, which ends up doing RnD because the scientists actually responsible for it don't want to and maybe upgrades a machine every now and then. 

The point is, Xenobio even in its current state is leaps and bounds more productive than those jobs, and given enough time I could think of plenty others that could use some real purpose far more than this.

**Comment by FudgePucker** [comment]
So peridaxon is too powerful to have outside of bruteforcing xenobotany and would lead to someone doing xenobio every round to produce it reliably, but "instead of making things that can be acquired through better methods, have slime cores produce entirely new things people will want."

Sometimes there isn't a miner or mining borg, how are you going to get that single sheet to copy using the mat synth then? You can do it with xenobio now.

Maybe also instead of having simple easy to make chems like bicardine and dermaline, have unique chems like black food dye and 'nothing' available.

**Comment by 9600bauds** [comment]
> the chem that makes surgery completely unnecessary

It fixes internal organs and that's it. That's livers and lungs in practical terms. It doesn't fix broken bones, it doesn't fix internal bleeding, it doesn't swap heads.

**Comment by icantthinkofanameritenow** [comment]
@FudgePucker

> Sometimes there isn't a miner or mining borg, how are you going to get that single sheet to copy using the mat synth then?

You do what I always do- grab a Ripley or a pickaxe from Xenoarchaeology, mine the ore, use the mining ore processor in the mining outpost (all of one or two hacked doors away if there isn't an AI to act as a doorknob), and collect your sheets. A dozen times faster than playing the RNG roulette with slimes, and all with stuff which is available to you at roundstart. Hell, if you really want to you can do RnD to make the ore processor yourself if you don't want to use the one in the mining outpost. Oh, and don't forget about the 30 plasma sheets already in the research outpost that nobody touches, as well as the gold in the vault and thel stacks of minerals that appear as a xenoarch find every now and then. Of course, all that assumes you somehow can't make a borg and order it to go mine for you. You do have a posibrain right in Robotics ready to be activated, after all. 

Incidentally, the QM and Cargo Technicians have mining access too so they can also go and mine. The latter usually doesn't have anything better to do other than hoard crates anyway, you'd be doing them a favor. 

@9600bauds
It also fixes damage to other internal organs, and organ damage is still one of the most common reasons for  surgery.

**Comment by 9600bauds** [comment]
> and organ damage is still one of the most common reasons for surgery. 

It is not, and internal organs besides lungs and liver never have any damage.

**Comment by icantthinkofanameritenow** [comment]
@9600bauds
Not entirely true. I've seen damaged hearts and kidneys too. Lungs and.livers are just the most commonly damaged organs, that's all.

**Comment by DenkouZ** [comment]
@PJB3005 Thank you for the input on the code. I will edit it this afternoon

@icantthinkofanameritenow If you have said it in the thread then I have no doubt debated you on the points you have brought up, so lets try talking again. As someone who frequently plays xenobio and plays borer when I can't be bothered actually playing, I am very well aware of how powerful Peridaxon. I am also aware that it can not replace physical surgery like @9600bauds  stated that it can not repair bones and it can not reattach heads or limbs, nor can it replace missing organs or repair mechanical ones. Also if you look at most of the people that come to surgery it is mostly people with broken bones from explosions or monster attacks and busted lungs, so still on the point a drip feed of Peridaxon couldn't hope to keep up with, but could ease the amount of people walking around coughing up blood becasue they don't want to go though surgery for something that can be prevented with an iron pill and ignoring some text. As you stated later it would be produced at a laughably slow rate and at such a small quantity that even if it could do all of those, it still wouldn't be able to keep up with surgery.

When dealing with materials you could do as you stated and go out and mine your own with a pickaxe then access mining by breaking and entering or with AI assistance. I have taken this into account. That is why I have taken the idea of putting it in as metal slimes that are an available mutation from grey slimes. So if there are no miners then whoever is in xenobio could possibly make some materials to get medbay their equipment and make a hyper cell or two if the borgs want it. Even more still that even if there isn't anyone in xenobio and you need the materials you could go down to the lab and throw some monkeys at the slimes at the chance of getting an easy quick mutation for materials.  I have mostly designed it in mind to be enough to access if the crew is under staffed but only enough to be a kick start if there are people willing to deliver on the promise of materials. Even using your idea it can work with itself. You can still finish RnD, ask xenobio for 2 sheets of diamond, print a drill and then go to the roid yourself and do a much more quicker and effective job.

Here you have made a good suggestion and one that I would gladly impliment with Black Dye and @FudgePucker also suggested 'Nothing', I hadn't considered these because I have never played bartender and I had never heard of 'nothing' before. These would fit very well into the pyrite slime ideas of producing service materials. You also state it would be a good thing for them to be able to produce mythril and adamantine because they are unobtainable but I would refute this because they have no real use at all. I have looked though the list of materials and even considered this while working on this PR and decided against it because there is no actual point to these existing even if you could make or process them. I am for the idea of Black Dye and 'nothing' because at least I can find some things they are used in. 

I am not sure if you play xenobiology because I have talked with some people that I see play the job over constant rounds and I have asked them ideas on what they would think of the the idea of the additional slime management potions. They all garnered very positive reactions and people in the thread came up with interesting ideas in the thread of mobile slime farms. If you think that these potions would be worthless with some small tweaks to the slime's base code I would like to see some examples. Becuase the same could be said for the extract enhancer and the slime steroid for that case. It was made to be an addition to slime management for those who were able to do the research for it.

**Comment by DenkouZ** [comment]
Also I do not state the production of universal enzyme because I saw this before submitting my PR
https://github.com/d3athrow/vgstation13/pull/8811. I know it is not merged but the point still stands.

**Comment by coldcola** [comment]
I'm not sure it's been mentioned but what's the deal with the slime producing already refined ores?
Could they possibly not do so to add a step in that makes them not as strong?
Just a question, I understand the entire idea here is to make xenobio useful if mining doesn't exist/is hanging out in the bar.

**Comment by icantthinkofanameritenow** [comment]
@DenkouZ

I have indeed, and quite frankly when I play it I don't give a crap about how useful it is to the station, I just breed the slimes. Comparing the potions to the existing ones is a false equivalence, since those directly affect the cores and the extracts and not the slimes themselves. 

> It was made to be an addition to slime management for those who were able to do the research for it.

The problem is that there is nothing to research. Dress it up all you like, at the end of the day Xenobiology is still just praying to the RNG not to leave you with trash. It would be better if you could directly influence what colors are produced from a split, rather than simply brute-force it by making them split more.

Speaking of which, I believe the relevant code is in here:
https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/modules/mob/living/carbon/slime/life.dm

If I'm reading this right, the variables that handle mutation chance and how quickly slimes grow start at line 327. Change those, and it'll translate to a flat increase in both growth speed and mutation rate.

**Comment by icantthinkofanameritenow** [comment]
Also, so what if mythril and adamantite do nothing? What's stopping you from coding things in which they can be used in, other than laziness?
The least you can do is copypaste the code that allows the creation of metal tiles, walls, and doors to them so you can make pretty rooms out of them or something. 

You gotta start thinking outside the box instead of adding a chance to MAYBE get a tiny amount of materials if you're lucky and/or don't get glomped to death. In fact, I would argue that the biggest reason so few people play Xenobio is because it is literally based on luck above all else. Remove the element of random chance where it doesn't contribute anything, and everything else will fall into place.

**Comment by DenkouZ** [comment]
@coldcola I am basing this idea off of how metal slimes produce processed plasteel and metal sheets, and how dark purple slime produce processed plasma sheets.
If that is a balance adjustment people want made then it would be fine by me. As @icantthinkofanameritenow said, RnD could always make their own sheets from ore.

@icantthinkofanameritenow I have considered this and looked at this dm while working on the PR. There are 2 issues with editing those. Slimes only require one monkey to have their amount_grown stat to reach 10. I don't see how this can be lowered further. Also increasing the mutation chance leads to a string of problems. One is that if you increase the amount that slimes mutate already, then you will struggle not to mutate the slimes at all when you do want to produce 20 bottles of frost oil (for what ever reason. I saw someone trying to do a plan with it once). Second of all it would break the rarity of the later slime mutations. At the moment you have about a ~3% chance to have a slime mutate successfully each time to reach a end tier slime from a grey slime. Statistically that is about 1 in 30 rounds. That is about once every two days. Depending on how much you increase that. It could jump very quickly to people pumping out golems by 12:35 once every 10 shifts. Removing the element of chance from xenobio turns it into R&D2. 

So what if mythril and adamantite do nothing? The point is that this is an expansion on the idea of slimes, not materials. I would like to see things come from other materials that are not implemented. But that is not the point of this PR and has actually nothing to do with the balance of it.

File: code/modules/mob/living/carbon/slime/slime.dm [hunk 4]
```diff
@@ -744,7 +745,19 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
...
+			qdel(O)
+			O = null
+
+		//slime res
+		if(istype(O, /obj/item/weapon/slimeres))
+			if(Uses == 0)
+				to_chat(user, "<span class='warning'>The solution doesn't work on used extracts!</span>")
+				return ..()
+			to_chat(user, "You splash the Slime Resurrection Serum onto the extract causing it to quiver and come to life.")
+			var/mob/living/carbon/slime/S = new primarytype
```
Comment (PJB3005) [inline code/modules/mob/living/carbon/slime/slime.dm]: You can set the `loc` directly by passing it as an argument to new:

``` DM
var/mob/living/carbon/slime/S = new primarytype(get_turf(src))
```

Could you go and do that for all the other news?

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 5]
```diff
@@ -1432,6 +1560,31 @@ datum
...
+		slimematerials
+			name = "Slime Materials"
+			id = "m_mats"
+			result = null
+			required_reagents = list("carbon" = 5)
+			result_amount = 1
+			required_container = /obj/item/slime_extract/silver
+			required_other = 1
+			on_reaction(var/datum/reagents/holder)
+				var/obj/item/stack/sheet/S1 = pick(new /obj/item/stack/sheet/plasteel,
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Recipes.dm]: Due to the way it is coded BYOND might actually be dumb and create a new item for every possible item but only use 1, this code might be cleaner:

``` DM
var/obj/item/stack/sheet/S1 = new pick(/obj/item/stack/sheet/plasteel,
    /obj/item/stack/sheet/metal,
    /obj/item/stack/sheet/mineral/plasma,
    /obj/item/stack/sheet/mineral/silver,
    /obj/item/stack/sheet/mineral/gold,
    /obj/item/stack/sheet/mineral/uranium
)
```

Comment (PJB3005) [inline code/modules/reagents/Chemistry-Recipes.dm]: On another note: copy pasting to do it twice is not a good idea, using a `for` loop is much cleaner:

``` DM
for(var/x = 1 to 2)
    var/obj/item/stack/sheet/S = ...
    S.amount = 5
    ...
```

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 6]
```diff
@@ -1681,6 +1884,28 @@ datum
...
+			on_reaction(var/datum/reagents/holder)
+				feedback_add_details("slime_cores_used","[replacetext(name," ","_")]")
+				if (!istype(holder.my_atom.loc,/obj/item/weapon/grenade/chem_grenade))
+					send_admin_alert(holder, reaction_name="oil slime + blood (Slime Genocide)")
+				else
+					send_admin_alert(holder, reaction_name="oil slime + blood (Slime Genocide) in a grenade")
+				for(var/mob/living/carbon/slime/S in viewers(get_turf(holder.my_atom), null)) // kills slimes
+					S.death(0)
+				for(var/mob/living/simple_animal/slime/S in viewers(get_turf(holder.my_atom), null)) // kills pet slimes too
+					S.health = 0
```
Comment (PJB3005) [inline code/modules/reagents/Chemistry-Recipes.dm]: I don't think this is the correct way to go about killing simple animals.

**Comment by DenkouZ** [comment]
@PJB3005 I am working on getting the code tidied up now. I will get in contact on the IRC later after my day closes up.

Could I also get someone to maybe stick a Update Wiki and Needs Changelog tag on here too maybe?

**Comment by Kammerjunk** [comment]
I also don't think that's the correct way to kill simple animals. I'm not sure what the actual way to kill them is, but IIRC, death(0) didn't work. I kinda forgot all about that, my bad.

**Comment by Probe1** [comment]
I skimmed over it but anything will do.  Xenobio needs love.

By skim I meant I read the whole thing but I didn't check the code and I didn't think about it in depth.  I didn't see anything repulsively imbalanced.  There may be issues down the road but there's nothing objectionable at this time.

**Comment by MusgardTiger** [comment]
I've read everything through, I am all for it. 

*clap p good features pal.

**Comment by icantthinkofanameritenow** [comment]
And I guess we can wait and see how it actually works in practice before I grumble further. Still, I wouldn't hold my breath about this making people think Xenobio is a waste of time and effort. At best, you'll have people rush adamantine slimes for diamond and then never come back. 

It would be a good idea to consider a long-term solution that reduces the randomness factor in Xenobio without completely eliminating it. 

My idea would be to introduce a special slime type that can rarely mutate from (and to)any other color, whose extracts would allow you to change the color of any other slime (but not to the color of this hypothetical slime, obviously). It would serve both as a reward for being so dedicated to slime breeding and as a way to accelerate the mutation process from there.  I'm thinking a rainbow color would be appropriate, don't you?

**Comment by ghost** [comment]
:+1:

**Comment by DenkouZ** [comment]
@icantthinkofanameritenow I am actually all for this idea. It would mean having to make a new slime type and adding it somewhere to the mutation tree that I am unsure of where it would sit considering how it sounds like it would expand beyond the current final tier of slimes. I will brainstorm some ideas later in a couple of weeks when I have more time on my hands and try coding some things then once more more ideas come into the mix. Also it would require a nice sprite too.

**Comment by DenkouZ** [comment]
@PJB3005 I managed to get most of that code you asked for tidied up and manageable. Some of it I couldn't sort though. I bugged Cluster and Probe for a while over the last couple of days. If there is something you want me to edit into the code. If you could show me an example I will see what I can do. Other than that I think the PR is in its final form.

**Comment by icantthinkofanameritenow** [comment]
@DenkouZ
I'd think it would work best as a separate if-then statement before the normal list of potential mutations- the game would roll up a number as a threshold and compare it to a value generated by the slime when it splits. If that number is less than or equal to the threshold, one of the slimes becomes a rainbow slime, otherwise the normal mutation rules apply. This way, it would theoretically be able to be found from any slime color mutating, albeit much less often than a "normal" color. 

I guess we'd want the threshold to be low enough so there's only a 5% chance of a rainbow slime appearing with each split. Could be lowered if needed though, since these things are meant to be rare by design.

This would also need to apply to the rainbow slimes themselves to prevent them from being bred en masse. Now that i think of it, that would create an interesting predicament, too: do you harvest the rainbow slime's core immediately, or do you take a gamble and see if it splits into a color you want? (Or will you be like me and use a docility potion on it and keep it as a pet because you can?)

**Comment by DenkouZ** [comment]
Okay. I fixed one small issue where the SlimeRes potion had the same reaction as the 'Make all slimes feral' reaction. Added a changelog, Wiki edit is written and waiting for the PR to merge.

**Comment by TheFriggenSawbones** [comment]
As a coding super-star I say we should add this now as it's good, and cool.

**Comment by icantthinkofanameritenow** [comment]
I don't know about you, but so far it seems like this update hasn't even come close to making xenobiology useful for the station. If anything, I get the impression that dedicated xenobiologists are the only ones who even give a shit about it since no regular science players who haven't been doing this exclusively have come lining up to play xenobio since. And just as I predicted, all the "new options" may as well not exist because they are either completely ignored or passed over in favor of doing things the traditional way.

**Comment by gbasood** [comment]
@icantthinkofanameritenow You're welcome to refactor xenobio into something both fun and useful if you want

But you better overhaul the entire system while you're at it

**Comment by Probe1** [comment]
tbh I don't know what the problem is, xenobios had someone playing it every round since it was added.    People are enjoying playing xenobio and that's ok in my book
